packages/@aws-cdk/cfnspec/lib/schema/property.ts

```diff
@@ -173,6 +178,14 @@ export function isMapProperty(prop: Property): prop is MapProperty {
   return (prop as MapProperty).Type === 'Map';
 }
 
+// note: this (and the MapOfListsOfPrimitives type) are not actually valid in the CFN spec!
+// they are only here to support our patch of the CFN spec
+// to alleviate https://github.com/aws/aws-cdk/issues/3092
+export function isMapOfListOfPrimitivesProperty(prop: Property): prop is MapOfListsOfPrimitives {
+  return isMapProperty(prop) && (prop as ComplexMapProperty).ItemType === 'List' &&
+    !!(prop as PrimitiveMapProperty).PrimitiveItemType;
```

Comment:
I don't think we should use `PrimitiveItemType` for this.

The rule right now is:

* If `Type=Map` then the item type is in either `ItemType|PrimitiveItemType`.

Right now you've put the type in both, and depending on how write our `ifs` that could be misinterpreted in some other part of the code.

Can we not invent a new field for this that can't be confused with the first-level generics? I would take any of:

* `PrimitiveItemItemType`
* `PrimitiveItemType2`
* `InnerPrimitiveItemType`
*  `MapListPrimitiveItemType`

Or come up with something better. But overloading the existing field seems dangerous to me (plus just conceptually wrong).

-----

This is how I justify `PrimitiveItemItemType`. Yes it sounds stupid but it is extremely regular:

```
    String
    Foo
    List              < Foo                            >
    List              < List                  < Foo  > >

    ^^^^                ^^^^^                   ^^^^
    [Primitive]Type     [Primitive]ItemType     [Primitive]ItemItemType
```

Reply:
Done. Went with `PrimitiveItemItemType`.